### PR TITLE
First stab at Rust BPF

### PR DESCRIPTION
**Pull request by jackcmay**
Issue tracking the effort: https://github.com/solana-labs/solana/issues/2268

### CODE REVIEW

File: ci/test-stable-perf.sh [hunk 1]
```diff
@@ -8,17 +8,20 @@ annotate() {
...
-ci/affects-files.sh \
-  .rs$ \
-  ci/test-stable-perf.sh \
-  ci/test-stable.sh \
-|| {
-  annotate --style info --context test-stable-perf \
-    "Stable Perf skipped as no .rs files were modified"
-  exit 0
-}
+# ci/affects-files.sh \
```
Comment (mvines) [inline ci/test-stable-perf.sh]: Let's fix this before landing please

File: programs/bpf/rust/noop/src/solana_sdk.rs [hunk 2]
```diff
@@ -0,0 +1,211 @@
+//! @brief Solana Rust-based BPF program utility functions and types
```
Comment (garious) [inline programs/bpf/rust/noop/src/solana_sdk.rs]: Looks like this module needs a better home and lots of unit tests

Comment (jackcmay) [reply programs/bpf/rust/noop/src/solana_sdk.rs]: Yup, mentioned as part of the 'needs to happen' stuff here: 
https://github.com/solana-labs/solana/issues/2268.  Do you want that to happen before landing this PR?

File: ci/test-stable-perf.sh (hunk 1 above)
Comment (jackcmay) [reply ci/test-stable-perf.sh]: Good catch!

**Comment by jackcmay** [comment]
@aeyakovenko 
Limitations and general notes can be found in #2268.  In general the limitations are that we can't use anything that doesn't get inlined and what does or does not get inlined is hard to quantify.  Once support added for linking in libraries we can link in the libs we need (like Rusts Core lib).  Check out #2268, I'm looking forward to comments and suggestions.

File: programs/bpf/rust/noop/src/solana_sdk.rs (hunk 2 above)
Comment (garious) [reply programs/bpf/rust/noop/src/solana_sdk.rs]: Yes, unit tests should come in alongside new code (or prior!)

Comment (jackcmay) [reply programs/bpf/rust/noop/src/solana_sdk.rs]: :-). Added now

**Pull request merged**